File: src/main/java/org/example/eshop/service/NotificationService.java
package org.example.eshop.service;

import org.example.eshop.entity.Order;
import org.example.eshop.repository.OrderItemRepository;
import org.example.eshop.repository.VariantRepository;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Service;

@Service
public class NotificationService {
    private static final Logger logger = LoggerFactory.getLogger(NotificationService.class);

    private final OrderItemRepository orderItemRepository;
    private final VariantRepository variantRepository;

    public NotificationService(OrderItemRepository orderItemRepository, VariantRepository variantRepository) {
        this.orderItemRepository = orderItemRepository;
        this.variantRepository = variantRepository;
    }

    public void logOrderReceived(Order order) {
        try {
            var orderItems = orderItemRepository.findByOrderId(order.getId());

            StringBuilder items = new StringBuilder();
            orderItems.forEach(orderItem -> items.append("- ")
                    .append(orderItem.getTitleSnapshot())
                    .append(" (Qty: ")
                    .append(orderItem.getQty())
                    .append(", Price: ")
                    .append(orderItem.getPriceSnapshot())
                    .append(")\n"));

            String logMessage = "" +
                    "ORDER RECEIVED - AWAITING PAYMENT\n" +
                    "Order Number: " + order.getNumber() + "\n" +
                    "Order Date: " + order.getCreatedAt() + "\n\n" +
                    "Items:\n" + items + "\n" +
                    "Financial Summary:\n" +
                    "Subtotal: " + order.getSubtotal() + "\n" +
                    "VAT Amount: " + order.getTax() + "\n" +
                    "Shipping Cost: " + order.getShipping() + "\n" +
                    "Total: " + order.getTotal() + "\n\n" +
                    "Payment Status: " + order.getPaymentStatus() + "\n" +
                    "Order Status: " + order.getStatus();

            logger.info(logMessage);
        } catch (Exception e) {
            logger.error("Failed to log order received event for order " + order.getNumber(), e);
        }
    }

    public void logPaymentReceived(Order order) {
        try {
            String logMessage = "" +
                    "PAYMENT RECEIVED\n" +
                    "Order Number: " + order.getNumber() + "\n" +
                    "Customer Email: " + order.getEmail() + "\n" +
                    "Payment Date: " + order.getUpdatedAt() + "\n" +
                    "Total Paid: " + order.getTotal() + "\n" +
                    "Payment Status: " + order.getPaymentStatus();
            logger.info(logMessage);
        } catch (Exception e) {
            logger.error("Failed to log payment received event for order " + order.getNumber(), e);
        }
    }

    public void logOrderShipped(Order order) {
        try {
            String tracking = order.getTrackingUrl() != null ? order.getTrackingUrl() : "Not provided";
            String logMessage = "" +
                    "ORDER SHIPPED\n" +
                    "Order Number: " + order.getNumber() + "\n" +
                    "Customer Email: " + order.getEmail() + "\n" +
                    "Ship Date: " + order.getUpdatedAt() + "\n" +
                    "Tracking URL: " + tracking + "\n" +
                    "Fulfillment Status: " + order.getFulfillmentStatus();
            logger.info(logMessage);
        } catch (Exception e) {
            logger.error("Failed to log order shipped event for order " + order.getNumber(), e);
        }
    }
}

File: src/main/kotlin/org/example/eshop/service/NotificationService.kt
package org.example.eshop.service

import org.example.eshop.entity.Order
import org.example.eshop.repository.OrderItemRepository
import org.example.eshop.repository.VariantRepository
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service
import java.math.BigDecimal

@Service
class NotificationService(
    private val orderItemRepository: OrderItemRepository,
    private val variantRepository: VariantRepository
) {
    
    private val logger = LoggerFactory.getLogger(NotificationService::class.java)
    
    fun logOrderReceived(order: Order) {
        try {
            val orderItems = orderItemRepository.findByOrderId(order.id)
            
            val itemDetails = orderItems.map { orderItem ->
                "- ${orderItem.titleSnapshot} (Qty: ${orderItem.qty}, Price: ${orderItem.priceSnapshot})"
            }.joinToString("\n")
            
            val logMessage = """
                |ORDER RECEIVED - AWAITING PAYMENT
                |Order Number: ${order.number}
                |Order Date: ${order.createdAt}
                |
                |Items:
                |$itemDetails
                |
                |Financial Summary:
                |Subtotal: ${order.subtotal}
                |VAT Amount: ${order.tax}
                |Shipping Cost: ${order.shipping}
                |Total: ${order.total}
                |
                |Payment Status: ${order.paymentStatus}
                |Order Status: ${order.status}
            """.trimMargin()
            
            logger.info(logMessage)
            
        } catch (e: Exception) {
            // Log the error but don't block order creation
            logger.error("Failed to log order received event for order ${order.number}", e)
        }
    }
    
    fun logPaymentReceived(order: Order) {
        try {
            val logMessage = """
                |PAYMENT RECEIVED
                |Order Number: ${order.number}
                |Customer Email: ${order.email}
                |Payment Date: ${order.updatedAt}
                |Total Paid: ${order.total}
                |Payment Status: ${order.paymentStatus}
            """.trimMargin()
            
            logger.info(logMessage)
            
        } catch (e: Exception) {
            logger.error("Failed to log payment received event for order ${order.number}", e)
        }
    }
    
    fun logOrderShipped(order: Order) {
        try {
            val logMessage = """
                |ORDER SHIPPED
                |Order Number: ${order.number}
                |Customer Email: ${order.email}
                |Ship Date: ${order.updatedAt}
                |Tracking URL: ${order.trackingUrl ?: "Not provided"}
                |Fulfillment Status: ${order.fulfillmentStatus}
            """.trimMargin()
            
            logger.info(logMessage)
            
        } catch (e: Exception) {
            logger.error("Failed to log order shipped event for order ${order.number}", e)
        }
    }
}